src/platform_impl/apple/uikit/view.rs

```diff
@@ -314,6 +316,11 @@ declare_class!(
             let mtm = MainThreadMarker::new().unwrap();
             app_state::handle_nonuser_event(mtm, gesture_event);
         }
+
+        #[method(canBecomeFirstResponder)]
+        fn can_become_first_responder(&self) -> bool {
+            true
+        }
     }
 
     unsafe impl NSObjectProtocol for WinitView {}
```

Comment:
@madsmtm I was surprised that the order of the impl Traits here mattered, first I put the `impl UIKeyInput` and `impl UITextInputTraits` above `impl NSObjectProtocol` and I would get `failed to add protocol NSObject` when starting the app. Is that expected? I also couldn't find anything about order in the objc2 docs.

Reply:
Wow... I've been having trouble with this too, but I never considered that it could be the ordering (in hindsight it makes perfect sense, but at the time it just felt like a fluke) - thank you so much for figuring it out!

Have fixed it in https://github.com/madsmtm/objc2/commit/36b4587fc1840b4c6cc24dfcdb7587ecdd81117f, which I'll ship with the next major release (not worth a patch release by itself IMO).